api/app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use Illuminate\Routing\Controller;
+use App\Http\Resources\UserResource;
+use App\Http\Requests\UpdateUserRequest; 
+
+
+class UserController extends Controller
+{
+  public function index()
+  { 
+    return UserResource::collection(User::get());
+  }
+ 
+  public function show($user_id)
+  {
+    try {
+      $user = User::findOrFail($user_id);
+      return response()->json(['data' => $user], 200);
+    } catch (\Throwable) {
+      return response()->json([
+        'message' => "The requested resource could not be found.", 
+      ], 404);
+    }
+  }
+
```

Comment:
```suggestion
  public function show(User $user_id)
  {
    return response()-json($user);
  }

```
Use Route Model Binding the parameters. It will automatically throw an 404 error if the instance does not exist and 200 if the instance exist
https://laravel.com/docs/9.x/routing#route-model-binding

Reply:
can you elaborate more

Reply:
wait, I'll try this